lib/sshkit/backends/netssh.rb

```diff
@@ -123,7 +129,9 @@ def transfer_summarizer(action, options = {})
       end
 
       def execute_command(cmd)
+        cmd.options[:redacted] = true
         output.log_command_start(cmd)
+        cmd.options[:redacted] = false
```

Comment:
This code strikes me as being a bit brittle. We are setting an option purely to get a desirable side-effect, when a more direct approach would be more self-explanatory.

What if the Command object had a method to get a redacted version of itself, and that is what gets logged?

I am open to suggestions on the name, but one possibility is:

```ruby
output.log_command_start(cmd.with_sensitive_args_redacted)
```

The `with_sensitive_args_redacted` method would be responsible for returning a copy of the command with a different set of args.

```ruby
def with_sensitive_args_redacted
  new_args = args.map( ... ) # logic to replace redacted args goes here
  redacted_cmd = dup
  redacted_cmd.instance_variable_set (:@args, new_args)
  redacted_cmd
end
```